lib/backend/dynamo/dynamodbbk.go

```diff
@@ -548,63 +548,129 @@ func (b *Backend) Update(ctx context.Context, item backend.Item) (*backend.Lease
 	return backend.NewLease(item), nil
 }
 
-// GetRange returns range of elements
-func (b *Backend) GetRange(ctx context.Context, startKey, endKey backend.Key, limit int) (*backend.GetResult, error) {
-	if startKey.IsZero() {
-		return nil, trace.BadParameter("missing parameter startKey")
-	}
-	if endKey.IsZero() {
-		return nil, trace.BadParameter("missing parameter endKey")
-	}
+func (b *Backend) queryOutputPages(ctx context.Context, limit int, input *dynamodb.QueryInput) iter.Seq2[*dynamodb.QueryOutput, error] {
 	if limit <= 0 {
 		limit = backend.DefaultRangeLimit
 	}
 
-	result, err := b.getAllRecords(ctx, startKey, endKey, limit)
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
-	sort.Sort(records(result.records))
-	values := make([]backend.Item, len(result.records))
-	for i, r := range result.records {
-		values[i] = backend.Item{
-			Key:      trimPrefix(r.FullPath),
-			Value:    r.Value,
-			Revision: r.Revision,
-		}
-		if r.Expires != nil {
-			values[i].Expires = time.Unix(*r.Expires, 0).UTC()
-		}
-		if values[i].Revision == "" {
-			values[i].Revision = backend.BlankRevision
+	return func(yield func(*dynamodb.QueryOutput, error) bool) {
+		const defaultPageSize = 1000
+		var nextToken map[string]types.AttributeValue
+
+		totalCount := 0
+		for {
+			input.Limit = aws.Int32(int32(min(limit-totalCount, defaultPageSize)))
+
+			result, err := b.svc.Query(ctx, input)
+			if err != nil {
+				yield(nil, trace.Wrap(err))
+				return
+			}
+
+			nextToken = result.LastEvaluatedKey
+			if !yield(result, nil) {
+				return
+			}
+
+			if nextToken == nil {
+				return
+			}
+
+			totalCount += len(result.Items)
```

Comment:
To avoid a potential `Query` with a zero (or negative, if something is misbehaving) `Limit`:

```suggestion
			totalCount += len(result.Items)
			if totalCount >= limit {
				return
			}
```